stratum/hal/config/tofino_chassis_config_migrator.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string>
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_split.h"
+#include "gflags/gflags.h"
+#include "stratum/glue/init_google.h"
+#include "stratum/glue/logging.h"
+#include "stratum/hal/lib/common/common.pb.h"
+#include "stratum/lib/macros.h"
+#include "stratum/lib/utils.h"
+
+DEFINE_string(chassis_config_file, "",
+              "Path to chassis configuration to migrate.");
+
+namespace stratum {
+namespace hal {
+namespace config {
+
+const char kUsage[] =
+    R"USAGE(usage: --chassis_config_file=<path>
+
+Chassis configuration migration tool.
+
+Combine with xargs for mass migration:
+ls -1 stratum/hal/config/*/chassis_config.pb.txt | \
+  xargs -n 1 bazel run //stratum/hal/config:tofino_chassis_config_migrator -- \
+    -chassis_config_file
+)USAGE";
+
+::util::Status Main(int argc, char** argv) {
+  ::gflags::SetUsageMessage(kUsage);
+  InitGoogle(argv[0], &argc, &argv, true);
+  stratum::InitStratumLogging();
+
+  CHECK_RETURN_IF_FALSE(!FLAGS_chassis_config_file.empty())
+      << "No chassis config given.";
+  ChassisConfig config;
+  RETURN_IF_ERROR(ReadProtoFromTextFile(FLAGS_chassis_config_file, &config));
+  if (config.chassis().platform() != PLT_GENERIC_BAREFOOT_TOFINO &&
+      config.chassis().platform() != PLT_GENERIC_BAREFOOT_TOFINO2) {
+    RETURN_ERROR(ERR_INVALID_PARAM)
+        << "Chassis config is not for Tofino platform";
+  }
+  for (auto& singleton_port : *config.mutable_singleton_ports()) {
+    std::vector<std::string> parts = absl::StrSplit(singleton_port.name(), '/');
```

Comment:
This takes the port name as source of truth. We could also use the slot/port/channel combo, as it kinda must be correct.

Reply:
Can we use the value from the `port` and `channel` fields in the `singleton_ports` message instead of parsing the `name` field?

Reply:
Changed to code to do that.